map.go

```diff
@@ -1,917 +0,0 @@
-package xsync
-
-import (
-	"fmt"
-	"math"
-	"runtime"
-	"strings"
-	"sync"
-	"sync/atomic"
-	"unsafe"
-)
-
-type mapResizeHint int
-
-const (
-	mapGrowHint   mapResizeHint = 0
-	mapShrinkHint mapResizeHint = 1
-	mapClearHint  mapResizeHint = 2
-)
-
-const (
-	// number of Map entries per bucket; 3 entries lead to size of 64B
-	// (one cache line) on 64-bit machines
-	entriesPerMapBucket = 3
-	// threshold fraction of table occupation to start a table shrinking
-	// when deleting the last entry in a bucket chain
-	mapShrinkFraction = 128
-	// map load factor to trigger a table resize during insertion;
-	// a map holds up to mapLoadFactor*entriesPerMapBucket*mapTableLen
-	// key-value pairs (this is a soft limit)
-	mapLoadFactor = 0.75
-	// minimal table size, i.e. number of buckets; thus, minimal map
-	// capacity can be calculated as entriesPerMapBucket*defaultMinMapTableLen
-	defaultMinMapTableLen = 32
-	// minimum counter stripes to use
-	minMapCounterLen = 8
-	// maximum counter stripes to use; stands for around 4KB of memory
-	maxMapCounterLen = 32
-)
-
-var (
-	topHashMask       = uint64((1<<20)-1) << 44
-	topHashEntryMasks = [3]uint64{
-		topHashMask,
-		topHashMask >> 20,
-		topHashMask >> 40,
-	}
-)
-
-// Map is like a Go map[string]interface{} but is safe for concurrent
-// use by multiple goroutines without additional locking or
-// coordination. It follows the interface of sync.Map with
-// a number of valuable extensions like Compute or Size.
-//
-// A Map must not be copied after first use.
-//
-// Map uses a modified version of Cache-Line Hash Table (CLHT)
-// data structure: https://github.com/LPD-EPFL/CLHT
-//
-// CLHT is built around idea to organize the hash table in
-// cache-line-sized buckets, so that on all modern CPUs update
-// operations complete with at most one cache-line transfer.
-// Also, Get operations involve no write to memory, as well as no
-// mutexes or any other sort of locks. Due to this design, in all
-// considered scenarios Map outperforms sync.Map.
-//
-// One important difference with sync.Map is that only string keys
-// are supported. That's because Golang standard library does not
-// expose the built-in hash functions for interface{} values.
-type Map struct {
```

Comment:
So, `Map` is being deleted, it means we're OK breaking everyone still using it (if there even is anyone left). That's 100% fine by me. But if that's the case, maybe `MapOf` can be renamed to `Map`, since the `Of` suffix is a vestige of there being two implementations.

What we could do is rename `MapOf` to `Map`, but leave behind this:
```
// Deprecated: Use [Map]
type MapOf[K comparable, V any] = Map[K, V]

// Deprecated: Use [NewMap]
func NewMapOf[K comparable, V any](options ...func(*MapConfig)) *MapOf[K, V] {
	return NewMap(options...)
}

...
```
That way transitioning from v3 to v4 is easy, but these types would eventually be removed. WDYT?

Reply:
Basically, like this: https://github.com/puzpuzpuz/xsync/pull/164

I can finish the remaining types if you think this is a good idea!

Incidentally, this is actually only possible as of 1.24! Since [type aliases](https://github.com/golang/go/issues/46477) were not supported before then, so declaring this alias would have been a compile error:
```go
type MapOf[K comparable, V any] = Map[K, V]
```

Reply:
That's a good idea! Please feel free to go ahead with the remaining types.

Reply:
Done!